services_app/src/main/java/org/opendatakit/services/preferences/fragments/ServerSettingsFragment.java

```diff
@@ -223,8 +196,67 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
     }
 
     healthValidator.verifyTableHealth();
+    setHasOptionsMenu(true);
   }
 
+
+  public boolean urlPreferenceChanged(Preference preference, Object newValue){
+    String url = newValue.toString();
+    // disallow any whitespace
+    if (url.contains(" ") || !url.equals(url.trim())) {
+      Toast.makeText(getActivity().getApplicationContext(),
+              R.string.url_error_whitespace, Toast.LENGTH_SHORT)
+              .show();
+      return false;
+    }
+
+    // remove all trailing "/"s
+    while (url.endsWith("/")) {
+      url = url.substring(0, url.length() - 1);
+    }
+
+    boolean isValid = false;
+    try {
+      new URL(url);
+      isValid = true;
+    } catch (MalformedURLException e) {
+      // ignore
+    }
+
+    if (isValid) {
+      preference.setSummary(newValue.toString());
+      updatePreferenceSummary(CommonToolProperties.KEY_SYNC_SERVER_URL, newValue.toString());
+      return true;
+    } else {
+      Toast.makeText(getActivity().getApplicationContext(),
+              R.string.url_error, Toast.LENGTH_SHORT)
+              .show();
+      return false;
+    } 
+  }
+
+  public boolean signOnPreferenceChanged(Preference preference, Object newValue){
+    int index = ((ListPreference) preference).findIndexOfValue(newValue.toString());
+    String entry = (String) ((ListPreference) preference).getEntries()[index];
+    ((ListPreference) preference).setSummary(entry);
+    updatePreferenceSummary(CommonToolProperties.KEY_AUTHENTICATION_TYPE, newValue.toString());
+
+    return true;
+  }
+  @Override
+  public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+    inflater.inflate(R.menu.server_settings_action_menu, menu);
+  }
+  private void updatePreferenceSummary(String key, String value){
```

Comment:
This name doesn't reflect what the method does, maybe `updatePropertiesSingleton`?